pkg/exchange/ftx/stream.go

```diff
@@ -102,15 +117,69 @@ func (s *Stream) SetPublicOnly() {
 	atomic.StoreInt32(&s.publicOnly, 1)
 }
 
-func (s *Stream) Subscribe(channel types.Channel, symbol string, _ types.SubscribeOptions) {
-	if channel != types.BookChannel {
-		panic("only support book channel now")
+func (s *Stream) Subscribe(channel types.Channel, symbol string, option types.SubscribeOptions) {
+	if channel == types.BookChannel {
+		s.addSubscription(websocketRequest{
+			Operation: subscribe,
+			Channel:   orderBookChannel,
+			Market:    toLocalSymbol(TrimUpperString(symbol)),
+		})
+
+	} else if channel == types.KLineChannel {
+		// FTX does not support kline channel, do polling
+		go s.subscribeKLine(symbol, option)
+	} else {
+		panic("only support book/kline channel now")
 	}
-	s.addSubscription(websocketRequest{
-		Operation: subscribe,
-		Channel:   orderBookChannel,
-		Market:    toLocalSymbol(TrimUpperString(symbol)),
+}
+
+func (s *Stream) handleChannelKlineMessage(ctx context.Context) {
+	for {
+		select {
+		case km := <-s.km:
+			// FTX only returns closed kline
+			klines := getLastKLine(s.exchange, ctx, km.symbol, km.interval)
+			if len(klines) >= 0 {
+				// handle mutiple klines, get the latest one
+				s.EmitKLineClosed(klines[len(klines)-1])
+			}
+		}
+	}
+}
+
+func (s *Stream) subscribeKLine(symbol string, option types.SubscribeOptions) {
```

Comment:
can we rename this method to pollKLines?

Reply:
fixed